packages/plugin-ext/src/common/plugin-api-rpc.ts

```diff
@@ -538,6 +540,16 @@ export interface WorkspaceExt {
     $onTextSearchResult(searchRequestId: number, done: boolean, result?: SearchInWorkspaceResult): void;
 }
 
+export interface TimelineExt {
+    $getTimeline(id: string, uri: string, options: theia.TimelineOptions, internalOptions?: theia.TimelineOptions): Promise<Timeline | undefined>;
```

Comment:
what is `internalOptions`? There are never used, but I see that they are used in VS Code. Could you elaborate why we don't need it?

Reply:
Reworked the `timeline.ts` to be similar to the one from vscode as much as possible, applied the `internalOptions` from vscode